From 1dd5db26176813b8a7c35781cf4f85d9d6cf6f9c Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Sun, 11 Dec 2016 11:46:09 -0500
Subject: [PATCH 1106/1936] ExternalPrograms: Fix Race In Makefile

.fl -> .cxx,.h generation could occur while gcc is running
with parallel make prior to this commit
---
 ExternalPrograms/Controller/Makefile | 5 ++++-
 ExternalPrograms/Spliter/Makefile    | 7 +++++--
 2 files changed, 9 insertions(+), 3 deletions(-)

diff --git a/ExternalPrograms/Controller/Makefile b/ExternalPrograms/Controller/Makefile
index 549607fc..ff747fad 100644
--- a/ExternalPrograms/Controller/Makefile
+++ b/ExternalPrograms/Controller/Makefile
@@ -1,9 +1,12 @@
 controller: main.o Controller.o ControllerUI.o
 	gcc -o controller *.o `fltk-config --ldflags` -lasound -lpthread -lm -lstdc++
 
-ControllerUI.cxx ControllerUI.h: ControllerUI.fl
+ControllerUI.h: ControllerUI.fl
 	fluid -c ControllerUI.fl
 
+ControllerUI.cxx: ControllerUI.h
+	echo ControllerUI.cxx
+
 ControllerUI.o: ControllerUI.cxx ControllerUI.h
 	gcc `fltk-config --cflags` -c ControllerUI.cxx -o ControllerUI.o
 
diff --git a/ExternalPrograms/Spliter/Makefile b/ExternalPrograms/Spliter/Makefile
index fe610d4d..c98fc412 100644
--- a/ExternalPrograms/Spliter/Makefile
+++ b/ExternalPrograms/Spliter/Makefile
@@ -1,10 +1,13 @@
 spliter: main.o Spliter.o SpliterUI.o
 	gcc -o spliter *.o `fltk-config --ldflags` -lasound -lstdc++
 
-SplitterUI.cxx SpliterUI.h: SpliterUI.fl
+SpliterUI.h: SpliterUI.fl
 	fluid -c SpliterUI.fl
 
-SpliterUI.o: SplitterUI.cxx SpliterUI.h
+SpliterUI.cxx: SpliterUI.h
+	echo SpliterUI.cxx
+
+SpliterUI.o: SpliterUI.cxx SpliterUI.h
 	gcc -g `fltk-config --cflags` -c SpliterUI.cxx -o SpliterUI.o
 
 Spliter.o: Spliter.C Spliter.h
-- 
2.47.0

